File: sdk/java/examples/02_java_main_api/src/main/java/io/touca/examples/main/StudentsTest.java
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

package io.touca.examples.main;

import java.time.LocalDate;
import io.touca.Touca;

public final class StudentsTest {

  @Touca.Workflow
  public void findStudent(final String username) {
    Touca.startTimer("find_student");
    Student student = Students.findStudent(username);
    Touca.stopTimer("find_student");
    Touca.addAssertion("username", student.username);
    Touca.addResult("fullname", student.fullname);
    Touca.addResult("birth_date", student.dob);
    Touca.addResult("gpa", student.gpa);
    Touca.addMetric("external_source", 1500);
  }

  public static void main(String[] args) {
    Touca.addTypeAdapter(LocalDate.class, x -> x.toString());
    Touca.run(StudentsTest.class, args);
  }

}

File: sdk/js/examples/02_node_main_api/students_test.ts
// Copyright 2021 Touca, Inc. Subject to Apache-2.0 License.

import { touca } from '@touca/node';
import { find_student } from './students';

touca.workflow('students_test', async (username: string) => {
  touca.start_timer('find_student');
  const student = await find_student(username);
  touca.stop_timer('find_student');
  touca.add_assertion('username', student.username);
  touca.add_result('fullname', student.fullname);
  touca.add_result('birth_date', student.dob);
  touca.add_result('gpa', student.gpa);
  touca.add_metric('external_source', 1500);
});

touca.run();
